cmd/baton-okta/config.go

```diff
@@ -23,7 +27,7 @@ func validateConfig(ctx context.Context, cfg *config) error {
 		return fmt.Errorf("domain is missing")
 	}
 
-	if cfg.ApiToken == "" {
+	if cfg.OktaClientId == "" && cfg.ApiToken == "" {
```

Comment:
Technically, client id is optional because Okta auth will work without it if there's only one possible client, but we might as well require it to simplify the code.

Ideally we'd check for api token, and if that's missing we'd check for client id, private key, and private key id. If all are missing, we'd return a general error message like "either api token or client id is required". If client id is set but private key or private key id isn't, we'd return a different error.

Also we should probably error out if api token and client id are set. Make the user pick one or the other instead of defaulting to api key in that case.